vm/systemSmartContracts/validator.go

```diff
@@ -175,12 +191,22 @@ func NewValidatorSmartContract(
 		enableUnbondTokensV2Epoch:        args.EpochConfig.EnableEpochs.UnbondTokensV2EnableEpoch,
 		validatorToDelegationEnableEpoch: args.EpochConfig.EnableEpochs.ValidatorToDelegationEnableEpoch,
 		shardCoordinator:                 args.ShardCoordinator,
+		stakeLimitsEnableEpoch:           args.EpochConfig.EnableEpochs.StakeLimitsEnableEpoch,
+		nodeLimitPercentage:              args.StakingSCConfig.NodeLimitPercentage / 100.0,
+		nodesCoordinator:                 args.NodesCoordinator,
+	}
+
+	reg.totalStakeLimit = core.GetIntTrimmedPercentageOfValue(args.GenesisTotalSupply, args.StakingSCConfig.StakeLimitPercentage)
```

Comment:
Could the percentage be of the CirculatingSupply, not the GenesisTotalSupply? It would be compared to up to ~31 million in 9 years not to 20 million.

Reply:
I think it is enough to compute GenesisTotalSupply.

Reply:
ok